systemtests/battery_test.go

```diff
@@ -13,40 +12,70 @@ import (
 )
 
 func (s *systemtestSuite) TestBatteryMultiMountSameHost(c *C) {
-	c.Skip("Can't be run until docker fixes this bug")
+	outerCount := 5
 	count := 15
-	errChan := make(chan error, count)
 
-	c.Assert(s.createVolume("mon0", "policy1", "test", nil), IsNil)
-	dockerCmd := "docker run -d -v policy1/test:/mnt alpine sleep 10m"
-	c.Assert(s.vagrant.GetNode("mon0").RunCommand(dockerCmd), IsNil)
+	for i := 0; i < outerCount; i++ {
+		syncChan := make(chan struct{})
 
-	for x := 0; x < count; x++ {
-		go func() {
-			dockerCmd := "docker run -d -v policy1/test:/mnt alpine sleep 10m"
-			errChan <- s.vagrant.GetNode("mon0").RunCommand(dockerCmd)
-		}()
-	}
+		for x := 0; x < count; x++ {
+			go func(x int) {
+				defer func() { syncChan <- struct{}{} }()
+				c.Assert(s.createVolume("mon0", "policy1", fmt.Sprintf("test%d", x), nil), IsNil)
+				dockerCmd := fmt.Sprintf("run -d -v policy1/test%d:/mnt alpine sleep 10m", x)
+				out, err := s.docker(dockerCmd)
+				c.Assert(err, IsNil)
+				_, err = s.docker(dockerCmd)
+				c.Assert(err, NotNil)
+				_, err = s.mon0cmd(fmt.Sprintf("mount | grep rbd | grep -q policy1.test%d", x))
+				c.Assert(err, IsNil)
+				out2, err := s.docker(fmt.Sprintf("exec %s ls /mnt", strings.TrimSpace(out)))
+				c.Assert(err, IsNil)
+				c.Assert(strings.TrimSpace(out2), Equals, "lost+found")
+				out3, err := s.docker(fmt.Sprintf("rm -f %s", strings.TrimSpace(out)))
+				if err != nil {
+					log.Info(strings.TrimSpace(out3))
+				}
+				c.Assert(err, IsNil)
+			}(x)
+		}
 
-	var realErr error
+		for x := 0; x < count; x++ {
+			<-syncChan
+		}
 
-	for x := 0; x < count; x++ {
-		err := <-errChan
-		if err != nil {
-			realErr = err
+		// FIXME netplugin is broken
+		c.Assert(s.restartNetplugin(), IsNil)
```

Comment:
netplugin panics occasionally for some reason when I run this test; I hacked around it for now.